js/ui/list/ui.list.edit.strategy.grouped.js

```diff
@@ -144,44 +146,50 @@ var GroupedEditStrategy = EditStrategy.inherit({
         return index;
     },
 
-    getItemsByKeys: function(keys) {
-        var result = [];
+    getItemsByKeys: function(keys, items) {
+        var result = [],
+            dataSource = this._collectionWidget.getDataSource(),
+            group = dataSource && dataSource.group();
 
-        each(keys, function(_, key) {
-            var getItemMeta = function(groups) {
-                var index = this.getIndexByKey(key, groups);
-                var group = index && groups[index.group];
+        if(group) {
+            result = queryByOptions(query(items), { group: group }).toArray();
```

Comment:
Please, use "return" here. It will be shorter and let you avoid multiple changes